File: scripts/count-city-words.py
import re
from pathlib import Path

p = Path("src/content/geo/cityLongDescription.ts").read_text(encoding="utf-8")
for m in re.finditer(
    r'\n  (?:"([^"]+)"|([a-z][a-z0-9-]*)):\s*`([\s\S]*?)`,\s*(?=\n  |\n};)',
    p,
):
    key = m.group(1) or m.group(2)
    body = m.group(3)
    wc = len(body.split())
    print(f"{key}: {wc}")

File: scripts/count-city-words.mjs
import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const p = fs.readFileSync(
  path.join(__dirname, "../src/content/geo/cityLongDescription.ts"),
  "utf8",
);
const re =
  /\n  (?:"([^"]+)"|([a-z][a-z0-9-]*)):\s*`([\s\S]*?)`,\s*(?=\n  |\n};)/g;
let m;
while ((m = re.exec(p))) {
  const key = m[1] || m[2];
  const wc = m[3].trim().split(/\s+/).filter(Boolean).length;
  console.log(`${key}: ${wc}`);
}
